File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = nums[0];
        int answer = nums[0];
        for (int i = 1; i < nums.length; i++) {
            sum += nums[i];
            if (sum < nums[i]) {
                sum = nums[i];
            }
            answer = Math.max(sum, answer);
        }
        return answer;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maximum = nums[0]
        running = nums[0]
        for num in nums[1:]:
            if num > running + num:
                running = num
            else:
                running += num
            maximum = max(maximum, running)
        return maximum